client/js/splitpanelsection.js

```diff
@@ -0,0 +1,634 @@
+
+var _ = require('underscore')
+var $ = require('jquery')
+
+var SplitPanelSection = function(index, $panel, initData, parent) {
+
+    this.parent = parent;
+    this.listIndex = index;
+    this.name = $panel.attr("id");
+
+    if (this.name === undefined)
+        throw "All splitter panels require an id attribute";
+
+    $panel.css("position", "absolute");
+    $panel.css("box-sizing", "border-box");
+
+    this.panel = $panel;
+
+    this.minWidth = -1;
+    this.maxWidth = -1;
+    this.strongEdge = "none";
+    this.stretchyEdge = "none";
+    this.preferedWidth = -1;
+    this.level = 0;
+    this._visible = true;
+    this._lastWeakMinWidth = -1;
+    this.isStrongMinWidth === false
+    this._lastWeakMaxWidth = -1;
+    this.isStrongMaxWidth === false;
+    this._coreWidth = 0;
+    this._covered = false;
+    this._position = 0;
+    this.sectionOnTop = false;
+    this._stretchyNeighbours = "none";
+    this._stuckTo = { left: 0, right: 0, reset: function() { this.left = 0; this.right = 0; } };
+    this._nextSection = { };
+
+
+    this.initalise = function(initData) {
+        if (initData.minWidth !== undefined)
```

Comment:
This maybe a misunderstanding on my behalf. I have learned this wrong at the beginning. From what i can see it's not completely wrong but browsers don't enforce its readonlyness therefore not wise to use. I'll make this more robust.damoOn Feb 01, 2016, at 05:57 AM, Jonathon Love <notifications@github.com> wrote:In client/js/splitpanelsection.js:> +    this._visible = true;
> +    this._lastWeakMinWidth = -1;
> +    this.isStrongMinWidth === false
> +    this._lastWeakMaxWidth = -1;
> +    this.isStrongMaxWidth === false;
> +    this._coreWidth = 0;
> +    this._covered = false;
> +    this._position = 0;
> +    this.sectionOnTop = false;
> +    this._stretchyNeighbours = "none";
> +    this._stuckTo = { left: 0, right: 0, reset: function() { this.left = 0; this.right = 0; } };
> +    this._nextSection = { };
> +
> +
> +    this.initalise = function(initData) {
> +        if (initData.minWidth !== undefined)
is this kosher? i think you're relying on undefined not being undefined, but someone could always define it. undefined isn't a keyword. a safer option would be:if (typeof initData.minWidth !== 'undefined') {
orif (_.isUndefined(initData.minWidth)) {
—Reply to this email directly or view it on GitHub.